### ISSUE TITLE
The call-methylation  step error

### ISSUE THREAD

**Issue by HW-Superior** [hasindu2008/f5c#183]
Hi,
 The following error occurs when I used f5c to call methylation and I wonder how this can be resolved. The script as f5c call-methylation -t 40 -r pass.fq.gz -b psss.bam -g genome.chr.fa --pore r10 > meth.tsv.
[meth_main::INFO]ESC[1;34m Default methylation tsv output format is changed from f5c v0.7 onwards to match latest nanopolish output. Set --meth-o
ut-version=1 to fall back to the old format.ESC[0m
[init_core::INFO]ESC[1;34m builtin DNA R10 nucleotide model loadedESC[0m
[init_core::INFO]ESC[1;34m builtin DNA R10 cpg model loadedESC[0m
HDF5-DIAG: Error detected in HDF5 (1.14.3) thread 1:
  #000: H5D.c line 1061 in H5Dread(): can't synchronously read data
    major: Dataset
    minor: Read failed
  #001: H5D.c line 1008 in H5D__read_api_common(): can't read data
    major: Dataset
    minor: Read failed
  #002: H5VLcallback.c line 2092 in H5VL_dataset_read_direct(): dataset read failed
    major: Virtual Object Layer
    minor: Read failed
  #003: H5VLcallback.c line 2048 in H5VL__dataset_read(): dataset read failed
    major: Virtual Object Layer
    minor: Read failed
  #004: H5VLnative_dataset.c line 373 in H5VL__native_dataset_read(): can't read data
    major: Dataset
    minor: Read failed
  #005: H5Dio.c line 401 in H5D__read(): can't read data
    major: Dataset
    minor: Read failed
  #006: H5Dchunk.c line 2919 in H5D__chunk_read(): unable to read raw data chunk
    major: Low-level I/O
    minor: Read failed
  #007: H5Dchunk.c line 4548 in H5D__chunk_lock(): data pipeline read failed
    major: Dataset
    minor: Filter operation failed
  #008: H5Z.c line 1354 in H5Z_pipeline(): required filter 'vbz' is not registered
    major: Data filters
    minor: Read failed
  #009: H5PLint.c line 265 in H5PL_load(): can't find plugin. Check either HDF5_VOL_CONNECTOR, HDF5_PLUGIN_PATH, default location, or path set by
 H5PLxxx functions
    major: Plugin for dynamically loaded library
    minor: Object not found
  #010: H5PLpath.c line 804 in H5PL__find_plugin_in_path_table(): search in path /home/wanghao/anaconda3/envs/f5c/lib/hdf5/plugin encountered an 
error
    major: Plugin for dynamically loaded library
    minor: Can't get value
  #011: H5PLpath.c line 856 in H5PL__find_plugin_in_path(): can't open directory (/home/wanghao/anaconda3/envs/f5c/lib/hdf5/plugin). Please verif
y its existence
    major: Plugin for dynamically loaded library
    minor: Can't open directory or file
[fast5_read_multi_fast5::ERROR]ESC[1;31m The fast5 file is compressed with VBZ but the required plugin is not loaded. See https://f5c.bioinf.scie
nce/troubleshoot for instructions.
ESC[0m
[fast5_read_multi_fast5::WARNING]ESC[1;33m Failed to read raw data from dataset /read_931790ee-20c2-454d-9a49-39fa79a8a9cd/Raw/Signal.ESC[0m
[handle_bad_fast5::WARNING]ESC[1;33m Fast5 file [/home/HddData/YangSV/ONT/195A/1444_1D_PAG70083_ddda41ca/195A_fast5_pass//PAG70083_pass_ddda41ca_
3934e17e_1174.fast5] for read [931790ee-20c2-454d-9a49-39fa79a8a9cd] is unreadable and will be skippedESC[0m

**Comment by hasindu2008** [comment]
Hello,

The last part of the error message says:
```
The fast5 file is compressed with VBZ but the required plugin is not loaded. See https://f5c.bioinf.science/troubleshoot for instructions
```

Can you try the solution here: https://hasindu2008.github.io/f5c/docs/general-troubleshoot#the-fast5-file-is-compressed-with-vbz-but-the-required-plugin-is-not-loaded.

**Comment by HW-Superior** [comment]
Thank you very much! The error have been solved.


---------------- 

----------原邮件信息----------
发件人："Hasindu Gamaarachchi" ***@***.***>
收件人："hasindu2008/f5c" ***@***.***>
抄送人："Xiaohaohao" ***@***.***>, "Author" ***@***.***>
主题：Re: [hasindu2008/f5c] The call-methylation step error (Issue #183)
发送时间：2024-11-01 14:59:46

 
Hello,

The last part of the error message says:
```
The fast5 file is compressed with VBZ but the required plugin is not loaded. See https://f5c.bioinf.science/troubleshoot for instructions
```

Can you try the solution here: https://hasindu2008.github.io/f5c/docs/general-troubleshoot#the-fast5-file-is-compressed-with-vbz-but-the-required-plugin-is-not-loaded.

**Comment by HW-Superior** [comment]
Hi, I have another question. The meth-freq.tsv file I obtained from meth-freq only contains CG methylation information, but I also want to get the CHG and CHH sites information. I would like to ask if there are any parameters in the software that can be obtained.

**Comment by hasindu2008** [comment]
Hello,

Unfortunately, f5c only supports CpG 5-mc methylation. We have not got any models for the other contexts or types.

Thanks

**Comment by HW-Superior** [comment]
thanks, I wonder if there is any way to output the strand information to the result when running meth-freq step

**Comment by hasindu2008** [comment]
This is something that is currently not implemented in meth-freq unfortunately. At the moment, I would not have time to implement this feature. Are you looking for a one-off thing or for a routine workflow? If it is the latter, I could look into implementing when I get some time.

**Comment by HW-Superior** [comment]
I am building the analysis flow and need the direction information of the strand for subsequent analysis. Can I extract the strand information of the site from the input file, so that I can write a script to complete this, thank you!

**Comment by hasindu2008** [comment]
Yes, the output TSV from call-methylation (second column if I remember right) has the strand information.  I think this would be what you are lokking for?

**Comment by hasindu2008** [comment]
Feel free to reopen if the problem persists.
